### ISSUE TITLE
Examples: Text has insufficient color contrast

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31156]
## WCAG Level
Level AA

## Priority
Low

## Type
Colour

## Pages/screens/components affected
* [amp-user-notification](https://amp.dev/documentation/components/amp-user-notification/)

## Description
An important design consideration is to make sure there is enough contrast between the color of text and the color of its background. Good color contrast makes content more readable for people who find it difficult to see, or who have conditions that affect their ability to perceive color.

In the `amp-user-notification` examples, we encountered instances of text content with insufficient contrast.

## User impact
Without sufficient contrast, content can be difficult to perceive and therefore difficult to read. In extreme cases, content with low contrast may not be visible at all to people with color perception impairments.

## Required solution
Make sure the foreground and background colors used for text follow these rules:

* Standard text (including images of text, links, button labels, and other controls) has a contrast ratio of at least 4.5:1 against its background.
* Large text (18pt/24px or 14pt/18.5px bold) has a contrast ratio of at least 3:1.

Incidental text that is not part of an interactive element, is pure decoration, or is part of a logo or picture, is exempt from these rules.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

In the [`amp-user-notification` example on Glitch](https://daisy-purring-seal.glitch.me/amp-user-notification.html), the "Learn more" link uses blue text (`#0000EE`) on a dark teal background (`#46B6AC`), with a contrast ratio of 3.8:1.

![The user notification example, showing the blue 'Learn more' link](https://user-images.githubusercontent.com/54933393/99181051-c48f5b00-2723-11eb-92f2-48c4647c2a95.png)

We recommend darkening the foreground color for the link - for instance, simply using black (`#000`) to match the preceding text.

In the [`amp-user-notification` example in the documentation](https://amp.dev/documentation/examples/components/amp-user-notification/), the notification itself has been styled to have a semi-transparent background. As a result, depending on the content of the underlying page, the contrast of the text contained in the notification (which is currently styled to use the default color for text across the site, `#48525C`) can end up below the minimum 4.5:1 ratio.

![The semi-transparent user notification, when the underlying page contains dark content - in this case, the styled code blocks that use a dark background](https://user-images.githubusercontent.com/54933393/99181055-cbb66900-2723-11eb-9050-97c3d2c5810d.png)

```
amp-user-notification {
  ...
  background: hsla(0,0%,100%,0.7);
  ...
}
```

In this case, we recommend either making the foreground text content explicitly darker (e.g. `#000`), or increasing the opacity of the semi-transparent background.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check the contrast of all text using a tool like the [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/).
2. Check that the contrast level is at least 4.5:1 for small text (14pt/18.5px) or 3:1 for large text (18pt/24px or 14pt/18.5px bold)

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.4.3 Contrast (Minimum) (Level AA)](https://www.w3.org/TR/WCAG21/#contrast-minimum)

## More information
* [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/)
* [WebAIM: Color blindness](https://webaim.org/articles/visual/colorblind)

## Test data
Test date: November 2020
